### PR TITLE
dnn: merge tests from test_halide_layers to test_backends

### PR DESCRIPTION
**Pull request by fengyuentau**
Context: https://github.com/opencv/opencv/pull/24231#pullrequestreview-1628649980

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Review by dkurt — DISMISSED** [review]

**Comment by asmorkalov** [comment]
Some test fixes from https://github.com/opencv/opencv/pull/24196 are missing in the refactored code.

**Comment by asmorkalov** [comment]
FullyConnected.Accuracy test was modified. Batch option was removed.

**Comment by dkurt** [comment]
@asmorkalov, good catch!

**Comment by fengyuentau** [comment]
> Batch option was removed.

I dont see this change. Anything I missed?

File: modules/dnn/test/test_backends.cpp [hunk 1]
```diff
@@ -573,4 +573,935 @@ TEST_P(DNNTestNetwork, FastNeuralStyle_eccv16)
...
+        lInf = INF_ENGINE_VER_MAJOR_GE(2023000000) ? 0.016 : 7e-3;
+    }
+#endif
+    if (targetId == DNN_TARGET_CUDA_FP16)
+        l1 = 0.015;
+
+    testLayer(lp, input, backendId, targetId, false, l1, lInf);
+}
+
+INSTANTIATE_TEST_CASE_P(Layer_Test_Backends, FullyConnected, testing::Combine(
```
Comment (dkurt) [inline modules/dnn/test/test_backends.cpp:1034]: @fengyuentau, please copy from 4.x manually: https://github.com/opencv/opencv/blob/4.x/modules/dnn/test/test_halide_layers.cpp. There are still missed updates about batch: https://github.com/opencv/opencv/blob/9942757bab881811816478c86d56924cf35c0d22/modules/dnn/test/test_halide_layers.cpp#L480

Comment (fengyuentau) [reply modules/dnn/test/test_backends.cpp:1034]: Thanks, let me take a look!

Comment (fengyuentau) [reply modules/dnn/test/test_backends.cpp:1035]: I took a look on the recent changes on this file, and it looks like the `FullyConnected.Accuracy` is the only test case that got updated.

Comment (fengyuentau) [reply modules/dnn/test/test_backends.cpp:1035]: All tests are green now.

**Review by asmorkalov — APPROVED** [review]
:+1:

**Review by dkurt — APPROVED** [review]

**Pull request merged**